### PR TITLE
Updates go minimum version from go1.20 to go1.22.1

### PR DESCRIPTION
**Pull request by yoomee1313**
## Proposed changes

- This PR updates go minimum version from go1.20 to go1.22.1. Referenced https://github.com/BoostryJP/quorum/pull/76.
- This PR is built on https://github.com/klaytn/klaytn/pull/2119.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: crypto/ecies/ecies.go [hunk 1]
```diff
@@ -255,60 +259,52 @@ func symDecrypt(params *ECIESParams, key, ct []byte) (m []byte, err error) {
...
 	}
+
 	R, err := GenerateKey(rand, pub.Curve, params)
 	if err != nil {
-		return
+		return nil, err
 	}
 
-	hash := params.Hash()
+	// hash := params.Hash()
```
Comment (hyunsooda) [inline crypto/ecies/ecies.go]:
```suggestion

```

**Review by hyunsooda — DISMISSED** [review]

**Review by hyeonLewis — COMMENTED** [review]
How about replacing all deprecated `math.rand` together in this PR? Reference: https://github.com/ethereum/go-ethereum/pull/26710

**Comment by yoomee1313** [comment]
@hyeonLewis Good suggestion. The change's simple but size is quite big, so i'll apply it in another PR. https://github.com/klaytn/klaytn/compare/de3f2205b7097d3cc5fc23b3c2e6dfe69dd916d3...362868fb3a1623dbe823293f515a4ef028cc0aae

**Review by blukat29 — APPROVED** [review]

**Review by hyeonLewis — APPROVED** [review]

**Comment by blukat29** [comment]
@sjnam @JayChoi1736 PTAL

**Review by sjnam — APPROVED** [review]

**Pull request merged**